[Deploy command — 06] Deployment guide (#295)
software-mansion/protostar: website/docs/tutorials/guides/deployment.md

---
sidebar_label: Deployment
---

# Deployment

> Protostar `>=0.2.1` supports deploying smart contracts to a given network with the `protostar deploy` command. It has similar interface to `starknet deploy` command. [Read the CLI reference for the deploy command](/docs/cli-reference#deploy) to learn more about all supported arguments.

Reviewer: I think the information `>=0.2.1` is unnecesary, it's place is in changelog, not in a guide.

> Protostar `>=0.2.1` supports deploying smart contracts to a given network with the `protostar deploy` command. It has similar interface to `starknet deploy` command. [Read the CLI reference for the deploy command](/docs/cli-reference#deploy) to learn more about all supported arguments.

Reviewer: ```suggestion
Protostar `>=0.2.1` supports deploying smart contracts to a given network with the `protostar deploy` command. It has a similar interface to the `starknet deploy` command. [Read the CLI reference for the deploy command](/docs/cli-reference#deploy) to learn more about all supported arguments.
```

## Example — deploying the default contract
1. Create new project 
   ```bash
   $ protostar init
   ```
2. Build the project
   ```bash
   $ protostar build

>    ```

Reviewer: Should we include it in this section or linik init tutorial? Im not sure, your call

3. Deploy the contract to the testnet
   ```
   $ protostar deploy ./build/main.json --network alpha-goerli
   ```
    ```shell title="Deployment output"
    [INFO] Deploy transaction was sent.
    Contract address: 0x06a5ea9e42c921bd58e24b8da9d1fc91a488df0700b173f1c6bb0e453f68afec
    Transaction hash: 0x1cbba90ba0d1fbfba09b1f7a0f987134dd9a02a845ca89244b3272374d37ede

    https://goerli.voyager.online/contract/0x06a5ea9e42c921bd58e24b8da9d1fc91a488df0700b173f1c6bb0e453f68afec
    ```

## Using [configuration profiles](/docs/tutorials/project-initialization#configuration-profiles)

> Configuration profiles allow you to easily reuse configuration for devnet, testnet, and mainnet networks. You can do it by defining the following configuration profiles in the `protostar.toml` as demonstrated on the snippet below.

Reviewer: I think configuration profiles are more general and should have their own guide, it suggest they can't be used for other commands.

Reviewer: Ok, i missed that, can we link it here?
> If you want to know more about configuration profiles go HERE

or something similar

Reviewer: Nevermind, we already do ❤️

> Configuration profiles allow you to easily reuse configuration for devnet, testnet, and mainnet networks. You can do it by defining the following configuration profiles in the `protostar.toml` as demonstrated on the snippet below.

Reviewer: ```suggestion
Configuration profiles allow you to easily reuse configuration for devnet, testnet, and mainnet networks. You can definine sample configuration profiles in the `protostar.toml` as demonstrated on the snippet below.
```
